test-app/src/test/kotlin/uk/gov/justice/digital/hmpps/testapp/integration/auth/customizer/CustomTokenConverterTest.kt

```diff
@@ -0,0 +1,82 @@
+package uk.gov.justice.digital.hmpps.testapp.integration.auth.customizer
+
+import org.junit.jupiter.api.Test
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.context.TestConfiguration
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Import
+import uk.gov.justice.digital.hmpps.testapp.integration.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.testapp.resource.CustomAuthAwareAuthenticationConverter
+import uk.gov.justice.hmpps.kotlin.auth.dsl.ResourceServerConfigurationCustomizer
+
+@Import(CustomTokenConverterTest.CustomizerConfiguration::class)
+@SpringBootTest(
+  webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+  // allow overriding of the ResourceServerConfigurationCustomizer bean definition
+  properties = ["spring.main.allow-bean-definition-overriding=true"],
+)
+class CustomTokenConverterTest : IntegrationTestBase() {
+
+  private class CaseloadFinder {
+    fun find() = "MDI"
+  }
+
+  @TestConfiguration
+  class CustomizerConfiguration {
+    @Bean
+    fun resourceServerCustomizer() = ResourceServerConfigurationCustomizer {
+      // Add a custom token converter that gets caseload from "somewhere", e.g. could be a bean calling another service
+      oauth2 { tokenConverter = CustomAuthAwareAuthenticationConverter { CaseloadFinder().find() } }
+    }
```

Comment:
This test config demonstrates the new functionality - you can now tell the resource server to use your custom token converter.